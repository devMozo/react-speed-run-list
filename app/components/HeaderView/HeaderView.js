import PropTypes from 'prop-types';
import './HeaderView.scss';

const HeaderView = ({
 gameName, img, playerName, time, link,
}) => (
  <section className="HeaderView">
    <picture className="HeaderView__logo">
      <img src={img} alt={gameName} />
    </picture>
    <div className="HeaderView__description">
      <h2>{gameName}</h2>
      <p>{`1Â° Place: ${playerName}`}</p>
      <p>{`Time: ${time}`}</p>
    </div>
    <div className="HeaderView__link">
      <a href={link} rel="noopener noreferrer" target="_blank">
        Go Video
      </a>
    </div>
  </section>
);

HeaderView.propTypes = {
  gameName: PropTypes.string,
  img: PropTypes.string,
  playerName: PropTypes.string,
  time: PropTypes.string,
  link: PropTypes.string,
};

export default HeaderView;
