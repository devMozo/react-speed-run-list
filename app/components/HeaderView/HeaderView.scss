@import '../../fonts/index.scss';

.HeaderView {
  width: 700px;
  box-shadow: 0px 0px 40px 0px rgb(219, 157, 224);
  margin: 0 auto;

  font-family: $fontMedium;

  .HeaderView__logo,
  .HeaderView__description {
    display: inline-block;
    vertical-align: middle;
    padding: 20px;
  }

  .HeaderView__description {
    width: 400px;
  }

  .HeaderView__link {
    width: 100%;
    background-color: #333;
    text-align: center;
    transition: 0.3s;

    &:hover {
      opacity: 0.6;
    }

    a {
      width: 100%;
      line-height: 50px;
      display: inline-block;
      color: white;
      text-decoration: none;
    }
  }
}
