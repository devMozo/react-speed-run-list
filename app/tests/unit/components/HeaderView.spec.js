import React from 'react';
import { shallow } from 'enzyme';
import HeaderView from '../../../components/HeaderView/HeaderView';

describe('<HeaderView />', () => {
  const MOCKED_PROPS = () => ({
    gameName: "Mocked Game's Name",
    img: 'mocked.png',
    playerName: "Mocked Player's Name",
    time: '12345678',
    link: 'mocked.com',
  });

  it('should render correctly', () => {
    const component = shallow(<HeaderView {...MOCKED_PROPS()} />);

    expect(component).toMatchSnapshot();
  });

  it('it must contain the correct fields', () => {
    const props = MOCKED_PROPS();
    const component = shallow(<HeaderView {...props} />);
    const gameName = component.find('.HeaderView__description h2');
    const img = component.find('.HeaderView__logo img');
    const playerName = component.find('.HeaderView__description p').first();
    const time = component.find('.HeaderView__description p').last();
    const link = component.find('.HeaderView__link a');

    expect(gameName.text()).toEqual(props.gameName);
    expect(img.prop('src')).toEqual(props.img);
    expect(img.prop('alt')).toEqual(props.gameName);
    expect(playerName.text()).toContain(props.playerName);
    expect(time.text()).toContain(props.time);
    expect(link.prop('href')).toEqual(props.link);
  });
});
