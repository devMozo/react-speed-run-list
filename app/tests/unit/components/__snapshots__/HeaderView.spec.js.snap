// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderView /> should render correctly 1`] = `
<section
  className="HeaderView"
>
  <picture
    className="HeaderView__logo"
  >
    <img
      alt="Mocked Game's Name"
      src="mocked.png"
    />
  </picture>
  <div
    className="HeaderView__description"
  >
    <h2>
      Mocked Game's Name
    </h2>
    <p>
      1Â° Place: Mocked Player's Name
    </p>
    <p>
      Time: 12345678
    </p>
  </div>
  <div
    className="HeaderView__link"
  >
    <a
      href="mocked.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Go Video
    </a>
  </div>
</section>
`;
